### PR TITLE
Add link to ListTile replacement guide in layout error message

### LINKED ISSUE #48134: Make the ListTile tileWidth != leadingSize.width error assertion more informative
I encountered the same as in #37451, and #46048, but am specifically requesting we prevent the assertion from crashing the application. I think this behavior is internally inconsistent, as it is best compared to an overflow issue like in `Column` or `Row`, and should at most render the overflow warning. 

My understanding of the scenarios where this error occurs:
* widget mounts while screen is off - mentioned in #37451
* widget mounts while parent is collapsed  and/or hidden (my use case, https://github.com/flutter/flutter/issues/46048#issuecomment-570399892)

doctor info - would be surprised if it was non-universal though:
```
Flutter 1.12.12 • channel master • https://github.com/flutter/flutter.git
Framework • revision bae92c32e5 (6 weeks ago) • 2019-11-24 22:29:28 -0800
Engine • revision b6b54fd606
Tools • Dart 2.7.0
```

### PR DESCRIPTION
**Pull request by micimize**
## Description

Adds a link to the [custom widget replacement guide](https://api.flutter.dev/flutter/material/ListTile-class.html#material.ListTile.4) to `ListTile`'s `tileWidth != leadingSize.width` and `tileWidth != trailingSize.width` assertion messages.

## Related Issues

closes #48134

## Tests

I can't find any tests with "Please use a sized widget" so I believe the message text is untested anyways.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by micimize** [comment]
I guess @shihaohong would be the appropriate reviewer. Looks like `flutter-build` is broken for everyone?

**Review by shihaohong — APPROVED** [review]
LGTM.

**Comment by shihaohong** [comment]
@micimize `flutter-build` is broken when the flutter tip of tree is breaking a test. Until that is fixed, this cannot be merged. Adding the `waiting for tree to go green` label will land the change once the build has been fixed.

**Pull request merged**
